### ISSUE TITLE
Changing property within sub collections is not tracked

### ISSUE THREAD

**Issue by Symbai** [lostmsu/RecursiveChangeNotifier#2]
I have a base class like this:
```
class Node
{
	public ObservableCollection<Node> Children { get; } = new ObservableCollection<Node>();
}
```
then I got a master class like:
```
class Master : Node
{
	public Master()
	{
		listener = ChangeListener.Create(Children);
		listener.PropertyChanged += (s, e) => { IsDirty = true; };
		listener.CollectionChanged += (s, e) => { IsDirty = true; };
	}
}
```
And sub classes like:
```
class Slave : Node
{
	
}
```

The following changes are perfectly tracked:
```
var slave = new Slave();
Master.Children.Add(slave);
...
//Changing property of a sub class with a depth level of 1
slave.Boolean = true;
slave.Boolean = false;
...
//Adding sub classes to a sub class
var slaveB = new Slave();
slave.Children.Add(slaveB);
```

What is NOT tracked however changing a property with a depth level of 2 or lower.
```
var slaveB = new Slave();
slave.Children.Add(slaveB); //<---- tracked
slaveB.Boolean = true; //<--- NOT tracked
...
slaveB.Children.Add(slaveC); //<---- tracked
slaveC.Boolean = true; //<--- NOT tracked
```

**Comment by lostmsu** [comment]
@Symbai looks like you figured it out. What was the issue?

**Comment by Symbai** [comment]
First of all I've noticed that your original version works different when I was about to create a repro project. Your version does not even track child collection changes at all (I must have fixed that a great while ago in the source I had forgotten about when creating the issue report) so I closed the issue when I noticed I was using a custom version.

I also had changed your source on the subscribe method (https://github.com/lostmsu/RecursiveChangeNotifier/blob/00e99823a4ac98712e9d8613265bcc6da8ea0812/src/ChildChangeListener.cs#L47) to get properties of the base class it inherits as well and there were some issues in my changes to not keep track of multi nested classes, also added an RecursiveChangeNotifierInclude attribute to filter out unwanted properties.

All of this was necessary because  I had nested classes and I store the parent node as a public property within the child which in your version, caused a deadlock I  believe.

Anyway with all that custom made changes I felt not bothering you as we would probably mixed up stuff.
